terratorch/models/smp_model_factory.py

```diff
@@ -18,66 +99,222 @@ def build_model(
             msg = f"SMP models can only perform pixel wise tasks, but got task {task}"
             raise Exception(msg)
-        # backbone_kwargs = _extract_prefix_keys(kwargs, "backbone_")
+        
+        backbone_kwargs = _extract_prefix_keys(kwargs, "backbone_") # Encoder params should be prefixed backbone_
+        smp_kwargs = _extract_prefix_keys(backbone_kwargs, "smp_") # Smp model params should be prefixed smp_
+        aux_params = _extract_prefix_keys(backbone_kwargs, "aux_") # Auxiliary head params should be prefixed aux_
+
+        # Gets decoder module.
+        decoder_module = getattr(smp, decoder, None)
+        if decoder_module is None:
+            raise ValueError(f"Decoder {decoder} is not supported in SMP.")
+        
         if isinstance(pretrained, bool):
             if pretrained:
                 pretrained = "imagenet"
             else:
-                pretrained = None
-        if decoder == "unet":
-            model = smp.Unet(
-                encoder_name=backbone, encoder_weights=pretrained, in_channels=in_channels, classes=num_classes
-            )
+                pretrained = None  
+        
+        model_dict = None
+        # If encoder not currently supported by SMP (either Prithvi or custom encoder).
+        if backbone not in ENCODERS:
+            # These params must be included in the config file with appropriate prefix.
+            required_params = {
+                'encoder_depth': smp_kwargs,
+                'out_channels': backbone_kwargs,
+                'output_stride': backbone_kwargs
+            }
+            
+            for param, config_dict in required_params.items():
+                if param not in config_dict:
+                    raise ValueError(f"Config must include the '{param}' parameter")
+            # Can load model from local checkpoint. 
+            if "checkpoint_path" in backbone_kwargs:
+                
+                checkpoint_path = backbone_kwargs.pop("checkpoint_path")
+                print(f"Trying to load from the path defined in the config file, {checkpoint_path}.")
+
+                if self.CPU_ONLY:
+                    model_dict = torch.load(checkpoint_path, map_location="cpu")
+                else:
+                    model_dict = torch.load(checkpoint_path)        
+
+            if backbone.startswith("prithvi"): # Using Prithvi encoder (ViT or Swin).
+                backbone_class = self._make_smp_encoder(PrithviModelWrapper)
+                if  backbone.startswith("prithvi_swin"):
+                    backbone_kwargs['prithvi_class'] = MMSegSwinTransformer
+                elif backbone.startswith("prithvi_vit"):
+                    backbone_kwargs['prithvi_class'] = TemporalViTEncoder
+                else:
+                    msg = f"Prithvi Backbone not found."
+                    raise NotImplementedError(msg)
+            # Using new encoder (not Prithvi or SMP).
```

Comment:
using `timm.create_model` should handle all these complexities. `checkpoint_path` should also be something that we don't take, but that is passed to this constructor (See the info box [here](https://ibm.github.io/terratorch/models/))